reference/extensions/hooks.rst

```diff
@@ -1,4 +1,167 @@
 .. _reference_extensions_hooks:
 
 Hooks
------
\ No newline at end of file
+-----
+
+The Conan hooks is a feature intended to extend the Conan functionalities and let users customize the client behavior at determined
+points.
+
+Hook structure
+--------------
+
+A hook is a Python function that will be executed at certain points of Conan workflow
+to customize the client behavior without modifying the client sources or the recipe ones.
+In the :ref:`hooks reference <hooks_reference>` you can find the full list of hook functions
```

Comment:
This would be already the hooks reference, just the current section?

Reply:
Done, it has been incorporated to this same file.